segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -226,22 +223,33 @@ private void handleAppendResult(final Append append, Long newWriteOffset, Throwa
                         connection.send(dataAppendedAck);
                     }
                 }
+
+                if (append.getEventNumber() > state.getLowestFailedEventNumber()) {
```

Comment:
Should this be called while still under the ackLock above?

Reply:
`append` is immutable and `state.getLowestFailedEventNumber` can only change to a lower value. So whether we do it earlier (in the lock) or now, it will still fire under the same conditions. 

`WriterState` is thread safe so there is no need to sync on `getAckLock` (which is used only to make sure the acks are sent to the client in the correct order).